plugins/inputs/sqlserver/connectionstring.go

```diff
@@ -0,0 +1,100 @@
+package sqlserver
+
+import (
+	"net/url"
+	"strings"
+)
+
+const (
+	emptyServerName   = "<empty-server-name>"
+	emptyDatabaseName = "<empty-database-name>"
+)
+
+// getConnectionIdentifiers returns the sqlInstance and databaseName from the given connection string.
+// The name of the server is returned as-is in the connection string
+// If the connection string could not be parsed or sqlInstance/databaseName were not present, a placeholder value is returned
+func getConnectionIdentifiers(connectionString string) (sqlInstance string, databaseName string) {
+	if len(connectionString) == 0 {
+		return "", ""
+	}
+
+	trimmedConnectionString := strings.TrimSpace(connectionString)
+
+	if strings.HasPrefix(trimmedConnectionString, "odbc:") {
+		connectionStringWithoutOdbc := strings.TrimPrefix(trimmedConnectionString, "odbc:")
+		return parseConnectionStringKeyValue(connectionStringWithoutOdbc)
+	}
+	if strings.HasPrefix(trimmedConnectionString, "sqlserver://") {
+		return parseConnectionStringURL(trimmedConnectionString)
+	}
+	return parseConnectionStringKeyValue(trimmedConnectionString)
+}
+
+// parseConnectionStringKeyValue parses a "key=value;" connection string and returns the SQL instance and database name
+func parseConnectionStringKeyValue(connectionString string) (sqlInstance string, databaseName string) {
+	sqlInstance = ""
+	databaseName = ""
+
+	keyValuePairs := strings.Split(connectionString, ";")
+	for _, keyValuePair := range keyValuePairs {
+		if len(keyValuePair) == 0 {
+			continue
+		}
+
+		keyAndValue := strings.SplitN(keyValuePair, "=", 2)
+		key := strings.TrimSpace(strings.ToLower(keyAndValue[0]))
+		if len(key) == 0 {
+			continue
+		}
+
+		value := ""
+		if len(keyAndValue) > 1 {
+			value = strings.TrimSpace(keyAndValue[1])
+		}
+		if strings.EqualFold("server", key) {
+			sqlInstance = value
+			continue
+		}
+		if strings.EqualFold("database", key) {
+			databaseName = value
+		}
+	}
+
+	if sqlInstance == "" {
+		sqlInstance = emptyServerName
+	}
+	if databaseName == "" {
+		databaseName = emptyDatabaseName
+	}
+
+	return sqlInstance, databaseName
+}
+
+// parseConnectionStringURL parses a URL-formatted connection string and returns the SQL instance and database name
+func parseConnectionStringURL(connectionString string) (sqlInstance string, databaseName string) {
+	sqlInstance = emptyServerName
+	databaseName = emptyDatabaseName
+
+	u, err := url.Parse(connectionString)
+	if err != nil {
+		return sqlInstance, databaseName
```

Comment:
You _could_ use `return emptyServerName, emptyDatabaseName)` here to make clear you are returning the defaults, but as it is a matter of taste I leave this decision to you.

Reply:
Done.